#! /bin/sh /usr/share/dpatch/dpatch-run
## 25-soname-rpath-strip.dpatch by Jens Peter Secher <jps@debian.org>
##
## DP: Make it possible to override the stripping of the neko executable prior
## DP: to building rest of compiler tools and introduce a proper SONAME for
## DP: libneko.


@DPATCH@
diff -urNad neko-1.6.0~/Makefile neko-1.6.0/Makefile
--- neko-1.6.0~/Makefile	2007-11-27 20:01:09.000000000 +0100
+++ neko-1.6.0/Makefile	2007-11-27 20:53:53.000000000 +0100
@@ -4,12 +4,15 @@
 
 CFLAGS = -Wall -O3 -fPIC -fomit-frame-pointer -I vm -DCOMPACT_TABLE
 EXTFLAGS = -pthread
-MAKESO = $(CC) -shared -WBsymbolic
-LIBNEKO_NAME = libneko.so
+MAJOR=0
+MINOR=0
+MAKESO = $(CC) -shared -Wl,-soname,libneko.so.${MAJOR}
+LIBNEKO_NAME = libneko.so.${MAJOR}.${MINOR}
 LIBNEKO_LIBS = -ldl -lgc -lm
-NEKOVM_FLAGS = -Lbin -lneko
-STD_NDLL_FLAGS = ${NEKOVM_FLAGS}
+NEKOVM_FLAGS = -Lbin -lneko -Wl,-rpath -Wl,/usr/lib/neko
+STD_NDLL_FLAGS = ${NEKOVM_FLAGS} -lm
 INSTALL_FLAGS =
+STRIP=strip
 
 NEKO_EXEC = LD_LIBRARY_PATH=../bin:${LD_LIBRARY_PATH} NEKOPATH=../boot:../bin ../bin/neko
 
@@ -118,16 +121,19 @@
 
 bin/${LIBNEKO_NAME}: ${LIBNEKO_OBJECTS}
 	${MAKESO} ${EXTFLAGS} -o $@ ${LIBNEKO_OBJECTS} ${LIBNEKO_LIBS}
+	ln -s ${LIBNEKO_NAME} bin/libneko.so
+	ln -s ${LIBNEKO_NAME} bin/libneko.so.${MAJOR}
 
 bin/neko: $(VM_OBJECTS)
-	${CC} ${CFLAGS} ${EXTFLAGS} -o $@ ${VM_OBJECTS} ${NEKOVM_FLAGS}
-	strip bin/neko
+	${CC} ${CFLAGS} -o $@ ${VM_OBJECTS} ${NEKOVM_FLAGS}
+	$(STRIP) bin/neko
 
 bin/std.ndll: ${STD_OBJECTS}
 	${MAKESO} -o $@ ${STD_OBJECTS} ${STD_NDLL_FLAGS}
 
 clean:
 	rm -rf bin/${LIBNEKO_NAME} ${LIBNEKO_OBJECTS} ${VM_OBJECTS}
+	rm -rf bin/libneko.so bin/libneko.so.${MAJOR}
 	rm -rf bin/neko bin/nekoc bin/nekoml bin/nekotools
 	rm -rf bin/std bin/*.ndll bin/*.n libs/*/*.o
 	rm -rf src/*.n src/neko/*.n src/nekoml/*.n src/tools/*.n
